src/VecSim/vec_sim_tiered_index.h

```diff
@@ -98,37 +99,33 @@ class VecSimTieredIndex : public VecSimIndexInterface {
 
 private:
     virtual int addVectorWrapper(const void *blob, labelType label, void *auxiliaryCtx) override {
-        // Will be used only if a processing stage is needed
-        char processed_blob[this->backendIndex->getDataSize()];
-        const void *vector_to_add = this->backendIndex->processBlob(blob, processed_blob);
-        return this->addVector(vector_to_add, label, auxiliaryCtx);
+        char PORTABLE_ALIGN aligned_mem[this->backendIndex->getDataSize()];
+        const void *processed_blob = this->backendIndex->processBlob(blob, aligned_mem);
+        return this->addVector(processed_blob, label, auxiliaryCtx);
     }
 
     virtual VecSimQueryResult_List topKQueryWrapper(const void *queryBlob, size_t k,
                                                     VecSimQueryParams *queryParams) const override {
-        // Will be used only if a processing stage is needed
-        char processed_blob[this->backendIndex->getDataSize()];
-        const void *query_to_send = this->backendIndex->processBlob(queryBlob, processed_blob);
-        return this->topKQuery(query_to_send, k, queryParams);
+        char PORTABLE_ALIGN aligned_mem[this->backendIndex->getDataSize()];
+        const void *processed_blob = this->backendIndex->processBlob(queryBlob, aligned_mem);
+        return this->topKQuery(processed_blob, k, queryParams);
     }
 
     virtual VecSimQueryResult_List rangeQueryWrapper(const void *queryBlob, double radius,
                                                      VecSimQueryParams *queryParams,
                                                      VecSimQueryResult_Order order) const override {
-        // Will be used only if a processing stage is needed
-        char processed_blob[this->backendIndex->getDataSize()];
-        const void *query_to_send = this->backendIndex->processBlob(queryBlob, processed_blob);
+        char PORTABLE_ALIGN aligned_mem[this->backendIndex->getDataSize()];
+        const void *processed_blob = this->backendIndex->processBlob(queryBlob, aligned_mem);
 
-        return this->rangeQuery(query_to_send, radius, queryParams, order);
+        return this->rangeQuery(processed_blob, radius, queryParams, order);
     }
 
     virtual VecSimBatchIterator *
     newBatchIteratorWrapper(const void *queryBlob, VecSimQueryParams *queryParams) const override {
-        // Will be used only if a processing stage is needed
```

Comment:
return the comment? 😅

Reply:
I think that now it's a matter of luck, but the new name suggests better whether it will be used or not, don't you think?

Reply:
Maybe the comment should be on top of processed_blob